### PR TITLE
doc: add section for TypeScript types generation with GraphQL Code Generator

### PR DESCRIPTION
**Pull request by charlypoly**
Hi 👋🏼

I am Charly from The Guild, the current lead of the [GraphQL Code Generator](https://github.com/dotansimha/graphql-code-generator/) project.

We recently released a new way of generating TypeScript types from GraphQL Schema and documents that provide a very smooth experience (no extra type or code generation) and a simpler setup (typescript-based config, simpler options).

In this process, we thought you would be interested in our contribution to providing the best practices of TypeScript types generation documentation in URQL docs itself 🚀

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

File: docs/basics/typescript-integration.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+          )}
+        </ul>
+      )}
+    </div>
+  );
+}
+// ...
+```
+
+GraphQL Code Generator will help you by exposing helps to retrieve the data required by your component's Fragment.
```
Comment (saihaj) [inline docs/basics/typescript-integration.md:206]: we should reword this 🤔 
```suggestion
GraphQL Code Generator will help you by exposing types to retrieve the data required by your component's Fragment.
```
```suggestion
GraphQL Code Generator will help you by exposing helps to retrieve the data required by your component's Fragment.
```

File: docs/basics/typescript-integration.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    </div>
+  );
+}
+
+export default App;
+```
+
+_Examples with Vue are available [in the GraphQL Code Generator repository](https://github.com/dotansimha/graphql-code-generator/tree/master/examples/front-end/vue/urql)_.
+
+Using the generated `graphql()` function to write your GraphQL document results in instantly typed result and variables for queries, mutations and subscriptions!
```
Comment (saihaj) [inline docs/basics/typescript-integration.md:162]: maybe for `graphql()` we should link to codgen docs where we explain in depth what this does vs just tagged literal
```suggestion
Using the generated `graphql()` function to write your GraphQL document results in instantly typed result and variables for queries, mutations and subscriptions!
```

**Review by dotansimha — APPROVED** [review]

File: docs/basics/document-caching.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Document Caching
-order: 4
+order: 5
```
[the remark this answers is not in the document]
Comment (charlypoly) [reply docs/basics/document-caching.md]: Okay!
Applied in 3bb66cd8

File: docs/basics/typescript-integration.md (hunk 2 above)
Comment (charlypoly) [reply docs/basics/typescript-integration.md:162]: I'm not sure we need to explain how it works under the hood in the docs 👀 
Why not in a blog post?

File: docs/basics/document-caching.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (charlypoly) [reply docs/basics/document-caching.md]: @JoviDeCroock, [we do have a plugin for `graph-cache` types generator](https://www.npmjs.com/package/@graphql-codegen/typescript-urql-graphcache) (similar to server resolvers one)
I can work on a separate PR of it; what do you think?

**Review by JoviDeCroock — APPROVED** [review]

**Pull request merged**